### PR TITLE
Feature/spk 109. Create form flow: replace field specific dialog errors with in-line messages

### PR DESCRIPTION
**Pull request by AlexMisiulia**

### CODE REVIEW

File: app/src/main/java/com/simplifire/yotinfcclient/utils/ViewUtils.kt [hunk 1]
```diff
@@ -20,6 +20,20 @@ fun ViewGroup.inflateAdapterView(layoutId: Int): View {
     return LayoutInflater.from(context).inflate(layoutId, this, false)
 }
 
+fun TextView.setErrorRes(errorRes: Int?) {
```
Comment (cquemin) [inline app/src/main/java/com/simplifire/yotinfcclient/utils/ViewUtils.kt]: This should use a `@StringRes` annotation

File: app/src/main/java/com/simplifire/yotinfcclient/viewmodel/CreateFormDetailsVm.kt [hunk 2]
```diff
@@ -26,43 +26,40 @@ class CreateFormDetailsVm @Inject constructor(private val formDao: FormDao,
...
-    val fieldErrorLiveData: LiveData<Event<Int>>
-        get() = _fieldErrorLiveData
-
     private lateinit var currentUiFormState: UiForm
 
     fun saveField(rawField: UiFormCustomRawField) {
-        if(!validateField(rawField)) return
+        val validatedField = validateField(rawField)
+
+        val newField = if (!validatedField.hasErrors()) {
```
Comment (cquemin) [inline app/src/main/java/com/simplifire/yotinfcclient/viewmodel/CreateFormDetailsVm.kt]: This is slightly counter intuitive. You have a variable named `validatedField` (which comes as result of a method named `validateField `, which makes perfect sense. But then you checked if it has errors ? so it is not a validatedField if it does contain error right? and then the result of this block return a field.

I think validateField should also check internally if there are errors and return the correct Field type: raw or regular.

File: app/src/main/java/com/simplifire/yotinfcclient/utils/ViewUtils.kt (hunk 1 above)
Comment (AlexMisiulia) [reply app/src/main/java/com/simplifire/yotinfcclient/utils/ViewUtils.kt]: Nice suggestion, thanks! Done

File: app/src/main/java/com/simplifire/yotinfcclient/viewmodel/CreateFormDetailsVm.kt (hunk 2 above)
Comment (AlexMisiulia) [reply app/src/main/java/com/simplifire/yotinfcclient/viewmodel/CreateFormDetailsVm.kt]: Sounds good. Done

**Pull request merged**